guide/message-processor-service/message_processor.bal

```diff
@@ -0,0 +1,108 @@
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mb;
+
+documentation { Define the message queue endpoint for sending newly processed new information messages. }
+endpoint mb:SimpleQueueSender queueProcessedQueue {
+    host: "localhost",
+    port: 5672,
+    queueName: "ProcessedQueue"
+};
+documentation { Attributes associated with the service endpoint. }
+endpoint http:Listener httpListner {
+    port: 8080
+};
+documentation {  via HTTP/1.1. }
+@http:ServiceConfig {
+    basePath: "/"
+}
+service<http:Service> MessageProcessingService bind httpListner {
+    @http:ResourceConfig {
+        methods: ["POST"], consumes: ["application/json"], produces: ["plain/text"],
+        path: "/activity"
+    }
+    documentation { Resources for reciving geo activities captured from the sensor }
+    activity(endpoint conn, http:Request req) {
+        http:Response res = new;
+        json requestMessage = check req.getJsonPayload();
+        log:printInfo("Message resived at /activity: " + requestMessage.toString());
+        json responseMessage1 = { "Message": "Earth Quake of 10 Richter Scale at the loaction near Panama",
+            "AlarmStatus":
+            "ON", "DesasterRecoveryTeamStatus": "Dispatched" };
+        string msg1 = responseMessage1.toString();
+        mb:Message message1 = check queueProcessedQueue.createTextMessage(msg1);
+        var p = message1.setPriority(1);
+        _ = queueProcessedQueue->send(message1);
+        json responseMessage2 = { "AlarmStatus": "ON" };
+        string msg2 = responseMessage2.toString();
+        mb:Message message2 = check queueProcessedQueue.createTextMessage(msg2);
+        var p2 = message2.setPriority(2);
+        log:printInfo("Notifying Authority and then set off alarms" + msg2);
+        _ = queueProcessedQueue->send(message2);
+        res.setTextPayload("status: Recived and processed by Message Processor");
+        _ = conn->respond(res);
+    }
+    @http:ResourceConfig {
+        methods: ["POST"], consumes: ["application/json"], produces: ["application/json"],
+        path: "/health"
+    }
+    documentation { Resources for reciving geo activities captured from the sensor }
+    health(endpoint conn, http:Request req) {
+        http:Response res = new;
+        json requestMessage = check req.getJsonPayload();
+        log:printInfo("Message resived at /health: " + requestMessage.toString());
```

Comment:
"resived" --> "received"

You can directly call `requestMessage.getPayloadAsString()` to get the string representation instead of getting the `json` payload first, since you are not using it anywhere else?

Reply:
Fix the typo in all places.

Reply:
Remove unwanted redundant variables.

Reply:
Remove unwanted redundant variables.